gateway/src/main/java/org/georchestra/gateway/filter/headers/RemoveBasicAuthenticationHeaderFilterFactory.java

```diff
@@ -0,0 +1,39 @@
+package org.georchestra.gateway.filter.headers;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.http.HttpHeaders;
+
+/**
+ * This filter removes the "Authorization: basic ..." request headers from the
+ * request, but leaves the other types of "Authorization: ..." headers (e.g.
+ * bearer) untouched.
+ *
+ * geOrchestra does not manage Authorization headers other than the usual basic
+ * authentication. If a token / bearer is passed, then it is probably
+ * implementation specific to the proxified service and should not be altered.
+ *
+ */
+@Slf4j
+public class RemoveBasicAuthenticationHeaderFilterFactory extends AbstractGatewayFilterFactory<Object> {
+
+    @Override
+    public GatewayFilter apply(Object unused) {
+        return (exchange, chain) -> {
+            exchange.getRequest().mutate().headers(BasicAuthConsumer::removeMatching).build();
+            return chain.filter(exchange);
+        };
+    }
+
+    static class BasicAuthConsumer {
+        static void removeMatching(@NonNull HttpHeaders headers) {
+            headers.entrySet().stream()//
+                    .filter(k -> k.getKey().equalsIgnoreCase("Authorization")//
+                            && k.getValue().toString().startsWith("Basic "))//
```

Comment:
`k.getValue()` will return a `List<String>` (http headers can be multivalued), we still have to iterate on the possible multiple values.
Not sure on how to do that with java streams though ...

Also I think we have to ignore case on the `startsWith("Basic ")`.

Reply:
> http headers can be multivalued

and in most cases, we will have a list with only one String as value.

Reply:
> Also I think we have to ignore case on the `startsWith("Basic ")`.

actually not necessary (see test class where with different cases Basic/basic are tested).